Yoast/wordpress-seo
# Make sure paper blocks always have an ID

## Summary

This PR can be summarized in the following changelog entry:

* [not-userfacing] Fix wrong IDs on the paper block elements.

## Relevant technical choices:

- removes the IDs from some of the headings in the paper blocks: these IDs were using the paper _title_ which in some cases is made of multiple words, e.g.: "Author archives settings", "Date archives settings", etc.
- makes sure all the paper blocks receive a `paper_id` param, which is then used for the ID on the paper block

I've discussed a bit the IDs issue with @dariaknl and, while these IDs don't do anything functionally, agreed to keep them because they will very likely used for automated tests. Also agreed there should be a solid naming convention for IDs used for testing, but this is something to be addressed in the future.

## Test instructions
- go in the setting pages
- verify in the markup all the paper blocks have a correct ID attribute (tip: you can search in the code for the string `paper tab-block` which is the CSS class used on the paper blocks)
- verify the paper block headings don't have IDs any longer

Fixes #11640
